### PR TITLE
docs(bodiless-documentation): Redirect Aliases

### LINKED ISSUE #1345: [DOC] Redirect Aliases
## User Story
AS AN end user of the Bodiless documentation,
I WANT to understand how to edit page redirect aliases,
SO THAT, using the Editor Interface, I can configure redirects as needed by my site's requirements.

## Acceptance Criteria
1. The Documentation is updated to include how a user can configure page redirect aliases.

### PR DESCRIPTION
**Pull request by misterjones**
## Changes

- Add "Tools" section to "Content Editor Guide."
- Add "Tools Menu" page.
- Add "Redirect Aliases" page.
- Add screenshots.

## Related Issues

**Documents:**
- #1122 

---

Resolves #1345

### CODE REVIEW

File: packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+- The third value, _Status Code_, is optional, but, if provided, must be a number.
+- Ensure that you only have _single_ spaces delimiting your values; double-whitespaces are invalid.
+
+### Can't Edit Redirected Page
+
+If you want to edit a page that is being redirected, you will need to remove the associated redirect
+in order to reach the page, so that you are able to edit it.
+
+01. Remove the redirect alias.
+01. Refresh your edit environment.
```
Comment (hvanyo) [inline packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md:77]:
```suggestion
01. Restart your edit environment.
```
base on your note right above troubleshooting.  Refresh doesn't solve the issue.

File: packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+      "The redirects are not valid, please correct."
+      - For troubleshooting assistance, see: [Troubleshooting : Invalid
+        Redirects](#invalid-redirects).
+
+<!-- Inlining HTML to add multi-line info block with ordered list. -->
+<div class="warn">
+  <strong>Note:</strong>
+
+  - Added/Modified redirect aliases will take effect immediately after saving.
+  - For _removed_ redirect aliases to take effect, you must first restart your edit environment.
```
Comment (hvanyo) [inline packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md:56]: We don't have an ability to restart edit env.  So add a note that you must call your friendly developer and restart the edit environment.

File: packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+in order to reach the page, so that you are able to edit it.
+
+01. Remove the redirect alias.
+01. Refresh your edit environment.
+01. Go to the desired page, and make your edits.
+01. Re-add the redirect alias.
+
+### Maximum Call Stack Size Exceeded
+
+If you should accidentally create a redirect alias with the same URL for both the _From Path_ and
```
Comment (hvanyo) [inline packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md:83]: I think you should mention here as well "BodilessJS does not check for circular redirects and doesn't attempt to correct or provide a warning."

File: packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+/page-3/ /page-4/
+```
+
+If you want `/page-1/`, `/page-2/`, and `/page-3/` to all redirect to `/page-4/`, then you need to
+configure your redirect aliases as follows:
+
+```
+/page-1/ /page-4/
+/page-2/ /page-4/
+/page-3/ /page-4/
```
Comment (hvanyo) [inline packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md:111]: I would add another warning:
If you are adding/editing redirects via this editor and get circular or multiple redirect errors, please be aware that you site might have redirects configured elsewhere.  The developers have opportunity to put redirects at platform configuration files or within the CDN.  No matter where the redirects are placed they all must work together and not duplicate/contradict each other.

Comment (hvanyo) [inline packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md:111]: I would add another warning as well that there is no need for redirects of urls that differ with a trailing slash.  The platform should be handle a url without a trialing slash to url with a trailing automatically.

**Comment by Guilherme-Almeida-Zeni** [comment]
Should we mention somewhere in the documentation that, by adding the redirects to the form, the redirect list is saved in the src/data/site/redirect-aliases.json file?

**Comment by Guilherme-Almeida-Zeni** [comment]
Should we mention that it's possible to remove all redirects by saving the form empty?

File: packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md (hunk 1 above)
Comment (misterjones) [reply packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md]: Whoops — definitely meant to say "Restart." Good catch.

Fixed. 👍 
Added note about contacting developer.

See: [Fix](https://github.com/johnsonandjohnson/Bodiless-JS/pull/1346/commits/e57ab8fc1d3114907d9b26c584590a1be560c2d1#diff-f8448c57a79c474ad52a150c2ee5d347dd7f13b395539f67a47f28f8c05e2847R91-R92)

File: packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md (hunk 2 above)
Comment (misterjones) [reply packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md:69]: ✅  [Updated](https://github.com/johnsonandjohnson/Bodiless-JS/pull/1346/commits/e57ab8fc1d3114907d9b26c584590a1be560c2d1#diff-f8448c57a79c474ad52a150c2ee5d347dd7f13b395539f67a47f28f8c05e2847R70).

File: packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md (hunk 3 above)
Comment (misterjones) [reply packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md:98]: ✅  [Updated](https://github.com/johnsonandjohnson/Bodiless-JS/pull/1346/commits/e57ab8fc1d3114907d9b26c584590a1be560c2d1#diff-f8448c57a79c474ad52a150c2ee5d347dd7f13b395539f67a47f28f8c05e2847R100-R101).

File: packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md (hunk 4 above)
Comment (misterjones) [reply packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md:135]: ✅  Added **Note** boxes:

- [Here](https://github.com/johnsonandjohnson/Bodiless-JS/pull/1346/commits/e57ab8fc1d3114907d9b26c584590a1be560c2d1#diff-f8448c57a79c474ad52a150c2ee5d347dd7f13b395539f67a47f28f8c05e2847R105-R111); and
- [Here](https://github.com/johnsonandjohnson/Bodiless-JS/pull/1346/commits/e57ab8fc1d3114907d9b26c584590a1be560c2d1#diff-f8448c57a79c474ad52a150c2ee5d347dd7f13b395539f67a47f28f8c05e2847R138-R144)

Comment (misterjones) [reply packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md:135]: ✅  [Updated](https://github.com/johnsonandjohnson/Bodiless-JS/pull/1346/commits/e57ab8fc1d3114907d9b26c584590a1be560c2d1#diff-f8448c57a79c474ad52a150c2ee5d347dd7f13b395539f67a47f28f8c05e2847R50-R55).

**Comment by misterjones** [comment]
@Guilherme-Almeida-Zeni 

> Should we mention somewhere in the documentation that, by adding the redirects to the form, the redirect list is saved in the src/data/site/redirect-aliases.json file?

✅  [Updated](https://github.com/johnsonandjohnson/Bodiless-JS/pull/1346/commits/e57ab8fc1d3114907d9b26c584590a1be560c2d1#diff-f8448c57a79c474ad52a150c2ee5d347dd7f13b395539f67a47f28f8c05e2847R14-R17).

> Should we mention that it's possible to remove all redirects by saving the form empty?

✅  [Updated](https://github.com/johnsonandjohnson/Bodiless-JS/pull/1346/commits/e57ab8fc1d3114907d9b26c584590a1be560c2d1#diff-f8448c57a79c474ad52a150c2ee5d347dd7f13b395539f67a47f28f8c05e2847R48-R49).

File: packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+    - If your configuration is invalid, you will see the following:  
+      "The redirects are not valid, please correct."
+      - For troubleshooting assistance, see: [Troubleshooting : Invalid
+        Redirects](#invalid-redirects).
+
+<!-- Inlining HTML to add multi-line info block with ordered list. -->
+<div class="warn">
+  <strong>Note:</strong>
+
+  - Added/Modified redirect aliases will take effect immediately after saving.
```
Comment (TatsianaMiraniuk) [inline packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md:68]: The behavior is not always true for 'Modified' rules.
 It is true for the latest version of a rule, but not always true for a previous version of the rule. A couple of examples how it works:
Step 1: the rules below are added as the completely new rules.
```
/page-a/ /page-b/ 301
/cards/ /list2/ 301
```
Works as expected:
```
/page-a/ redirects to /page-b/ 301
/cards/ redirects to /list2/ 301
```

Step 2: the rules modified - /page-b/ is replaced by /page-c/ and /cards/ is replaced by /anchor/
```
/page-a/ page-c/ 301
/anchor/ /list2/ 301
```

As Results the following redirect happens
```
/page-a/ -> /page-c/
/cards/ -> /list2/
/anchor/ ->  /list2/
```
 /cards/ -> /list2/ still happens. 
So the previous versions of a modified rule in such way are considered as 'removed' and the redirects will happen until the Edit env restarts.

File: packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+### Maximum Call Stack Size Exceeded
+
+If you should accidentally create a redirect alias with the same URL for both the _From Path_ and
+the _To Path_, this will create a circular reference and result in a `RangeError` — "Maximum call
+stack size exceeded." BodilessJS does not check for circular redirects, and, therefore, does not
+make any attempt to correct them, nor provide any warning.
+
+Remove or correct the redirect alias causing the circular reference.
+
+?> **Note:** Redirects can be configured elsewhere.
```
Comment (TatsianaMiraniuk) [inline packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md:105]: @misterjones seems, ?> is a typo?

File: packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+If you want `/page-1/`, `/page-2/`, and `/page-3/` to all redirect to `/page-4/`, then you need to
+configure your redirect aliases as follows:
+
+```
+/page-1/ /page-4/
+/page-2/ /page-4/
+/page-3/ /page-4/
+```
+
+?> **Note:** Redirects can be configured elsewhere.
```
Comment (TatsianaMiraniuk) [inline packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md:138]: @misterjones seems, ?> is a typo?

File: packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+01. Re-add the redirect alias.
+
+### Maximum Call Stack Size Exceeded
+
+If you should accidentally create a redirect alias with the same URL for both the _From Path_ and
+the _To Path_, this will create a circular reference and result in a `RangeError` — "Maximum call
+stack size exceeded." BodilessJS does not check for circular redirects, and, therefore, does not
+make any attempt to correct them, nor provide any warning.
+
+Remove or correct the redirect alias causing the circular reference.
```
Comment (TatsianaMiraniuk) [inline packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md:103]: @misterjones should we remind here to restart the env after the changes?

**Review by TatsianaMiraniuk — CHANGES_REQUESTED** [review]

File: packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md (hunk 6 above)
Comment (misterjones) [reply packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md:118]: @TatsianaMiraniuk, the `?>` is an extended syntax for markdown, provided by Docsify — [General tips](https://docsify.js.org/#/helpers?id=general-tips).

You can see, on the [Redirect Aliases](https://pr-1346-ie5iwwa-jvaff4bu65vgm.us-2.platformsh.site/___docs/#/ContentEditorUserGuide/ToolsMenu/Aliases/) page, that they render as info/note blocks. 😄

File: packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md (hunk 7 above)
Comment (misterjones) [reply packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md:151]: @TatsianaMiraniuk, the `?>` is an extended syntax for markdown, provided by Docsify — [General tips](https://docsify.js.org/#/helpers?id=general-tips).

You can see, on the [Redirect Aliases](https://pr-1346-ie5iwwa-jvaff4bu65vgm.us-2.platformsh.site/___docs/#/ContentEditorUserGuide/ToolsMenu/Aliases/) page, that they render as info/note blocks. 😄

File: packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md (hunk 5 above)
Comment (misterjones) [reply packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md]: @TatsianaMiraniuk, ah, interesting.

Thank you for the thorough explanation! 😄 

✅  [Updated](https://github.com/johnsonandjohnson/Bodiless-JS/pull/1346/commits/8d5e043ff148543a03c6ec52c218045f3c49129b#diff-f8448c57a79c474ad52a150c2ee5d347dd7f13b395539f67a47f28f8c05e2847R71-R81).

File: packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md (hunk 8 above)
Comment (misterjones) [reply packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md]: @TatsianaMiraniuk, good call. 👍 

✅  [Updated](https://github.com/johnsonandjohnson/Bodiless-JS/pull/1346/commits/8d5e043ff148543a03c6ec52c218045f3c49129b#diff-f8448c57a79c474ad52a150c2ee5d347dd7f13b395539f67a47f28f8c05e2847R114-R116).

**Review by Guilherme-Almeida-Zeni — APPROVED** [review]

File: packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+
+01. While in [Edit Mode](../../#edit-mode), from the [Toolbar](../../#toolbar), click **Tools >
+    Aliases** to open the _Redirect Aliases_ form.  
+    ![Redirect Aliases form](./assets/ToolsRedirectAliases.jpg ':size=50%')
+01. In the textarea, edit/remove/add the aliases to satisfy the redirection configuration required
+    by your site.
+    - Each alias needs to be configured on a separate line within the textarea, represented as three
+      values delimited by a _single_ space:
+      - **From Path:** The old URL — a path on your site.
+        - If you don't provide one at the beginning of your path, a leading forward slash (`/`) will
```
[the remark this answers is not in the document]
Comment (misterjones) [reply packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md]: Hi @Guilherme-Almeida-Zeni, 

Yeah, wording on that was a bit clunky.
✅  [Fixed!](https://github.com/johnsonandjohnson/Bodiless-JS/pull/1346/commits/bbc6a6c094948b08b51049355ee0bb47796e4a7e#diff-f8448c57a79c474ad52a150c2ee5d347dd7f13b395539f67a47f28f8c05e2847R29-R30)

File: packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+## Troubleshooting
+
+### Invalid Redirects
+
+If you've received the message that your redirects are not valid, review the following items to help
+troubleshoot the issue:
+
+- The first two values, _From Path_ and _To Path_, are **required**, and cannot be numbers.
+- The third value, _Status Code_, is optional, but, if provided, must be a number.
+- Ensure that you only have _single_ spaces delimiting your values; double-whitespaces are invalid.
```
[the remark this answers is not in the document]
Comment (misterjones) [reply packages/bodiless-documentation/doc/ContentEditorUserGuide/ToolsMenu/Aliases/README.md:97]: @Guilherme-Almeida-Zeni, 

Good call. 👍 
✅  [Added note](https://github.com/johnsonandjohnson/Bodiless-JS/pull/1346/commits/bbc6a6c094948b08b51049355ee0bb47796e4a7e#diff-f8448c57a79c474ad52a150c2ee5d347dd7f13b395539f67a47f28f8c05e2847R61-R63).

**Review by Guilherme-Almeida-Zeni — APPROVED** [review]

**Review by hvanyo — APPROVED** [review]

**Review by TatsianaMiraniuk — APPROVED** [review]

**Pull request merged**